### PR TITLE
[ie/spreaker] Support podcast and feed pages

### LINKED ISSUE #10925: "Unable to extract show id" on https://www.spreaker.com/show/5887186/episodes/feed
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Japan

### Provide a description that is worded well enough to be understood

yt-dlp can't download podcast https://www.spreaker.com/show/5887186/episodes/feed

On the opposite, wget dowloads correctly the feed and each mp3 file found in that feed.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ ./yt-dlp -vU "https://www.spreaker.com/show/5887186/episodes/feed"
/Users/me/Library/Python/3.9/lib/python/site-packages/urllib3/__init__.py:35: NotOpenSSLWarning: urllib3 v2 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  warnings.warn(
[debug] Command-line config: ['-vU', 'https://www.spreaker.com/show/5887186/episodes/feed']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds [e8e6a982a] (zip)
[debug] Python 3.9.6 (CPython arm64 64bit) - macOS-14.6.1-arm64-arm-64bit (LibreSSL 2.8.3)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, mutagen-1.47.0, requests-2.32.3, sqlite3-3.43.2, urllib3-2.2.2, websockets-13.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1832 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.09.02.232855 from yt-dlp/yt-dlp-nightly-builds)
[SpreakerShowPage] Extracting URL: https://www.spreaker.com/show/5887186/episodes/feed
[SpreakerShowPage] 5887186: Downloading webpage
ERROR: [SpreakerShowPage] 5887186: Unable to extract show id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/Users/nicolasraoul/lab/orbinea/./yt-dlp/yt_dlp/extractor/common.py", line 740, in extract
    ie_result = self._real_extract(url)
  File "/Users/nicolasraoul/lab/orbinea/./yt-dlp/yt_dlp/extractor/spreaker.py", line 168, in _real_extract
    show_id = self._search_regex(
  File "/Users/nicolasraoul/lab/orbinea/./yt-dlp/yt_dlp/extractor/common.py", line 1333, in _search_regex
    raise RegexNotFoundError(f'Unable to extract {_name}')
```

### PR DESCRIPTION
**Pull request by subrat-lima**
### enhancement: added support for podcast and feed pages

This pull request contains the following updates:
1. Added support for feed pages. ex: [show-2317431](https://www.spreaker.com/show/2317431/episodes/feed)
2. Added support for podcast pages. ex: [classic-books](https://www.spreaker.com/podcast/classic-books--5336619)
3. Updated test output for existing Extractor

Note:
1. The page structures have been changed for some url patterns and would require further investigation before adding the fix and as I am not familiar with the site and so will take some time. I've added the note for the same.

Fixes #10925 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/spreaker.py [hunk 1]
```diff
@@ -156,17 +159,53 @@ def _real_extract(self, url):
...
+
+    def _real_extract(self, url):
+        show_id = self._match_id(url)
+        return self.url_result(
+            f'https://api.spreaker.com/show/{show_id}',
+            ie=SpreakerShowIE.ie_key(), video_id=show_id)
+
+
+class SpreakerFeedPageIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/show/(?P<id>[\d]+)/episodes/feed'
```
Comment (grqz) [inline yt_dlp/extractor/spreaker.py:198]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/show/(?P<id>\d+)/episodes/feed'
```

File: yt_dlp/extractor/spreaker.py [hunk 2]
```diff
@@ -156,17 +159,53 @@ def _real_extract(self, url):
 
 
 class SpreakerShowPageIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/show/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/show/(?P<id>[^/?#&]+)(?!/episodes/feed)'
```
Comment (grqz) [inline yt_dlp/extractor/spreaker.py:162]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?spreaker\.com/show/(?P<id>[^/?#&]+)(?:[?#]|$)(?!/episodes/feed)'
```

**Review by grqz — COMMENTED** [review]
fix valid url regex

File: yt_dlp/extractor/spreaker.py (hunk 1 above)
Comment (subrat-lima) [reply yt_dlp/extractor/spreaker.py]: thanks, I've updated the regex.

File: yt_dlp/extractor/spreaker.py (hunk 2 above)
Comment (subrat-lima) [reply yt_dlp/extractor/spreaker.py]: thanks I've updated the regex.

**Review by seproDev — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

**Pull request merged**